lib/queue.js

```diff
@@ -0,0 +1,118 @@
+'use strict'
+
+var async = require('async')
+var chalk = require('chalk')
+var debug = require('debug-log')('mashape-analytics-queue')
+var http = require('http')
+var pkg = require('../package.json')
+var util = require('util')
+
+/**
+ * ALF Queue
+ */
+var Queue = module.exports = function (serviceToken, environment, options) {
+  var self = this
+
+  this.opts = options
+  this.timeout = null
+  this.userAgent = util.format('%s/%s', 'mashape-analytics-agent-node', pkg.version)
+
+  // setup event queue
+  // TODO specify worker pool
+  // TODO use msgpack + gzip?
+  this.queue = async.queue(function (entry, done) {
+    // append entry to log
+    self.alf.har.log.entries.push(entry)
+
+    // start timer
+    if (!self.timeout) {
+      self.timeout = setTimeout(function () {
```

Comment:
the timeout should be set/cleared/reset outside the queue, every time the agent middleware function is called.